Sources/Preferences/Localization.swift

```diff
@@ -99,26 +89,42 @@ struct Localization {
 	- Parameter identifier: Identifier of the string to localize.
 	*/
 	static func get(identifier: Identifier) -> String {
-		let locale = Locale.current
-
 		// Force-unwrapped since all of the involved code is under our control.
 		let localizedDict = Localization.localizedStrings[identifier]!
 		let defaultLocalizedString = localizedDict["en"]!
 
-		guard
-			let languageCode = locale.languageCode,
-			let regionCode = locale.regionCode
-		else {
-			return defaultLocalizedString
+		// Iterate through all user-preferred languages until we find one that has a valid language code
+		var preferredLocale: Locale?
+		for localeID in Locale.preferredLanguages {
+			let locale = Locale(identifier: localeID)
+			if locale.languageCode != nil {
+				preferredLocale = locale
+				break
+			}
 		}
```

Comment:
This could be simplified to:

```swift
let preferredLocale = Locale.preferredLanguages
			.lazy
			.map { Locale(identifier: $0) }
			.first { $0.languageCode != nil }
			?? Locale.current
```

Reply:
The term "simplify" is *very* subjective in this context. 😄
I didn't do any performance comparisons, I just personally think my version is more readable.
Also, IIRC, `locale.languageCode` was never nil in my tests, so in the overwhelming majority of cases, the loop should stop after just one iteration anyway.